### PR TITLE
Remove gulp

### PR DESCRIPTION
**Pull request by jakubtobiasz**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | 2.0
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | n/a
| License         | MIT

This change will be taken into account while creating the new documentation.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,40 +13,18 @@
...
-    "gulp-order": "^1.2.0",
-    "gulp-sass": "^5.1.0",
-    "gulp-sourcemaps": "^3.0.0",
-    "gulp-uglifycss": "^1.1.0",
-    "merge-stream": "^2.0.0",
-    "rollup": "^2.79.0",
-    "rollup-plugin-terser": "^7.0.2",
     "sass": "^1.54.8",
     "sass-loader": "^13.0.0",
-    "upath": "^2.0.1",
```
Comment (diimpp) [inline package.json:50]: Just a thought out loud,
It's interesting, that gulp used [`upath`](https://www.npmjs.com/package/upath#why-), while webpack is fine with npm's `path` package.

File: package.json [hunk 2]
```diff
@@ -58,15 +35,11 @@
...
     "build:prod": "encore production",
-    "gulp": "gulp build",
     "lint": "yarn lint:js",
-    "lint:js": "eslint gulpfile.babel.js src/Sylius/Bundle/AdminBundle/gulpfile.babel.js src/Sylius/Bundle/ShopBundle/gulpfile.babel.js src/Sylius/Bundle/UiBundle/Resources/private/js src/Sylius/Bundle/AdminBundle/Resources/private/js src/Sylius/Bundle/ShopBundle/Resources/private/js",
     "postinstall": "semantic-ui-css-patch"
   },
   "repository": {
     "type": "git",
     "url": "git+https://github.com/Sylius/Sylius.git"
-  },
```
Comment (diimpp) [inline package.json:69]: Did you intend to remove `author` and `license` fields?
I'm not familiar with npm packages publishing rules, so don't know a possible reason.

File: phpunit.xml.dist [hunk 3]
```diff
@@ -24,5 +24,7 @@
 
         <server name="IS_DOCTRINE_ORM_SUPPORTED" value="true" />
         <server name="ESCAPE_JSON" value="true" />
+
+        <server name="KERNEL_CLASS" value="App\Kernel" />
```
Comment (diimpp) [inline phpunit.xml.dist]: Is this change related to gulp? Looks like should have been done at 1.13 branch.

File: psalm.xml [hunk 4]
```diff
@@ -147,6 +147,7 @@
                 <referencedMethod name="Symfony\Component\HttpKernel\Event\RequestEvent::isMasterRequest" />
                 <referencedMethod name="Symfony\Component\HttpKernel\Event\ExceptionEvent::isMasterRequest" />
                 <referencedMethod name="Symfony\Component\HttpKernel\Event\ResponseEvent::isMasterRequest" />
+                <referencedMethod name="Symfony\Component\Config\Definition\Builder\NodeDefinition::children" />
```
Comment (diimpp) [inline psalm.xml:150]: Is this change related to gulp? Looks like should have been done at 1.13 branch.

File: UPGRADE-2.0.md [hunk 5]
```diff
@@ -39,3 +39,7 @@
     * The following configuration options were removed:
 
         * `sylius.mailer.templates`
+
+## Frontend
+
+* `use_webpack` option was removed from the `sylius_ui` configuration, and the Webpack has become the only module bundler provided by Sylius.
```
Comment (diimpp) [inline UPGRADE-2.0.md:45]: It might be worth to explicitly mention removal of `use_webhook` twig global as anyone with extended templates might have it.

**Review by diimpp — APPROVED** [review]
Gulp to the dustbin of history! :grimacing: 

<s>P.S. I assume docs will be handled separately?</s>Sorry, I've managed to read PR description now :sweat_smile:
https://github.com/Sylius/Sylius/blob/104bc12d1bc8c7960d67ce1ec87b004eda574c3b/docs/cookbook/frontend/staying-with-gulp.rst#L52

File: package.json (hunk 2 above)
Comment (jakubtobiasz) [reply package.json:69]: These two entries were duplicated 😅

File: phpunit.xml.dist (hunk 3 above)
Comment (jakubtobiasz) [reply phpunit.xml.dist]: `1.13` is working, idk why the build broke here. I'll check that again, I'll consider moving these two changes into a separate PR.

File: psalm.xml (hunk 4 above)
Comment (jakubtobiasz) [reply psalm.xml:150]: 1.13 is working, idk why the build broke here. I'll check that again, I'll consider moving these two changes into a separate PR.

Comment (jakubtobiasz) [reply psalm.xml:150]: Ok, I remember. This one has appear after removing the `->booleanNode('use_webpack')->defaultTrue()->end()` line in the configuration. I've checked my PR adding this line, and it's an obvious bug of Psalm (I don't like it tbh). So yes, we might consider this change as a part of removing Gulp 🤣.

File: phpunit.xml.dist (hunk 3 above)
Comment (jakubtobiasz) [reply phpunit.xml.dist]: I also checked this, and this change is a part of making the test app working [`ef923f9` (#15314)](https://github.com/Sylius/Sylius/pull/15314/commits/ef923f915162966dc0417d5dcfaa8e3da22e90fa). For some reason before it worked, and after adding the Webpack Encore it stopped, so it might also be considered as an integral part of this PR.

Comment (jakubtobiasz) [reply phpunit.xml.dist]: I removed this line, while implementing Twig Components in Sylius I discovered the issue only exists when we try to test bundle's PHPUnit suite from the main vendor (not the package one) 😅.

**Review by Prometee — APPROVED** [review]

**Review by NoResponseMate — APPROVED** [review]
